uos_freespace/src/uos_freespace.cpp

```diff
@@ -74,9 +80,9 @@ int FreeSpace::isInvertedScannerCheck(const sensor_msgs::LaserScan::ConstPtr &la
 	try	{
 		//wait for the tf frames to be available then transform the laserscan frame to the base frame
 		//this makes it so all scanners data is read as right-side up
-		tf_.waitForTransform("base_link", "laser",laserscan->header.stamp,ros::Duration(1.0));
-		tf_.transformQuaternion("base_link", min_q, min_q);
-		tf_.transformQuaternion("base_link", max_q, max_q);
+		tf_.waitForTransform(tf_prefix + "base_link", tf_prefix + "laser", laserscan->header.stamp, ros::Duration(1.0));
+		tf_.transformQuaternion(tf_prefix + "base_link", min_q, min_q);
+		tf_.transformQuaternion(tf_prefix + "base_link", max_q, max_q);
```

Comment:
Please use tf::resolve() instead. This way you also don't need to manually check for trailing '/' above.